File: VB-Examples/19_PivotTables/PivotTableLayout.vb
Imports Spire.Xls
Imports Spire.Xls.Core.Spreadsheet.PivotTables
Imports System.ComponentModel
Imports System.Text

Namespace PivotTableLayout
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub btnRun_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnRun.Click
            ' Create a new Workbook object
            Dim workbook As New Workbook()

            ' Load the Excel file from a specified path
            workbook.LoadFromFile("..\..\..\..\..\..\Data\PivotTable.xlsx")

            ' Get the first worksheet in the workbook
            Dim worksheet As Worksheet = workbook.Worksheets(0)

            ' Retrieve the PivotTable object from the worksheet and cast it as an XlsPivotTable
            Dim xlsPivotTable As XlsPivotTable = CType(worksheet.PivotTables(0), XlsPivotTable)

            ' Set the report layout of the PivotTable to Tabular
            xlsPivotTable.Options.ReportLayout = PivotTableLayoutType.Tabular

            ' Specify the file name for the result file
            Dim result As String = "PivotLayoutTabular_output.xlsx"

            ' Save the workbook to the specified file path with the specified Excel version (2013)
            workbook.SaveToFile(result, ExcelVersion.Version2013)

            ' Release the resources used by the workbook
            workbook.Dispose()

            'View the document
            FileViewer(result)
		End Sub

		Private Sub FileViewer(ByVal fileName As String)
			Try
				Process.Start(fileName)
			Catch
			End Try
		End Sub

	End Class
End Namespace

File: CS-Examples/19_PivotTables/PivotTableLayout.cs
using Spire.Xls;
using Spire.Xls.Core.Spreadsheet.PivotTables;
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;

namespace PivotTableLayout
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void btnRun_Click(object sender, EventArgs e)
        {
            // Create a workbook
            Workbook workbook = new Workbook();

            // Load the file from disk
            workbook.LoadFromFile(@"..\..\..\..\..\..\Data\PivotTable.xlsx");

            // Get the first worksheet
            Worksheet worksheet = workbook.Worksheets[0];

            // Get the first PivotTable
            XlsPivotTable xlsPivotTable = (XlsPivotTable)worksheet.PivotTables[0];

            // Set the PivotTable layout type
            xlsPivotTable.Options.ReportLayout = PivotTableLayoutType.Tabular;

            // Save to file
            String result = "PivotLayoutTabular_output.xlsx";
            workbook.SaveToFile(result, ExcelVersion.Version2013);

            // Dispose of the workbook object to release resources
            workbook.Dispose();

            // View the document
            FileViewer(result);
        }

        private void FileViewer(string fileName)
        {
            try
            {
                System.Diagnostics.Process.Start(fileName);
            }
            catch { }
        }

    }
}
